src/psyclone/tests/psyad/transformations/test_preprocess.py

```diff
@@ -129,3 +129,31 @@ def test_preprocess_matmul(tmpdir, fortran_reader, fortran_writer):
     result = fortran_writer(psyir)
     assert result == expected
     assert Compile(tmpdir).string_compiles(result)
+
+
+def test_preprocess_arrayrange2loop(tmpdir, fortran_reader, fortran_writer):
+    '''Test that the preprocess script replaces assignments that contain
+    arrays that use range notation with equivalent code that uses
+    explicit loops.
+
+    '''
+    code = (
+        "program test\n"
+        "real :: a(10), b(10), c(10)\n"
+        "a(:) = b(:) * c(:)\n"
```

Comment:
Could you extend this test with an assignment that uses rank-2 arrays as well please.

Reply:
Thanks.